iree/compiler/Dialect/HAL/hal.imports.mlir

```diff
@@ -403,4 +403,50 @@ vm.import @executable_layout.create(
 ) -> !vm.ref<!hal.executable_layout>
 attributes {nosideeffects}
 
+//===----------------------------------------------------------------------===//
+// iree::hal::Semaphore
+//===----------------------------------------------------------------------===//
+
+// Returns a semaphore from the device pool with the given initial value.
+vm.import @semaphore.create(
+  %device : !vm.ref<!hal.device>,
+  %initial_value : i32
```

Comment:
Here it's just a storage type - it's really signless. I'm not actually sure since Lei's changes for signed types landed whether it's possible to define si32/ui32 or something. Will have to play with that (and fix up everything) at some point.